### ISSUE TITLE
llvm_asm: invalid operand for instruction since nightly-2021-01-09

### ISSUE THREAD

**Issue by jschwe** [gz/rust-x86#70]
Hello,

When building the project [uhyve](https://github.com/hermitcore/uhyve) with `RUSTFLAGS="-Clink-dead-code" ` I get a compile error using `nightly-2021-01-09` and `nightly-2021-01-10` (but not before that).
```
error: invalid operand for instruction
 --> /home/jschwe/.cargo/registry/src/github.com-1ecc6299db9ec823/x86-0.34.0/src/bits32/mod.rs:9:5
  |
9 |     llvm_asm!("mov esp, $0; jmp $1" :: "rg"(stack), "r"(ip) :: "volatile", "intel");
  |     ^
  |
note: instantiated into assembly here
 --> <inline asm>:2:2
  |
2 |     mov esp, rdi; jmp rsi
  |     ^

error: aborting due to previous error

error: could not compile `x86`
```

Is this an issue with the nightly, or did something change in llvm_syntax, which would require an update to the x86 crate?

**Comment by gz** [comment]
I can't reproduce this with either

cargo build
cargo build --target i686-unknown-linux-gnu

on my 64bit Linux machine. Can you give me some more info on your environment (OS, rustc version)?

I tried these two rustc versions:
```
rustc --version                                      
rustc 1.51.0-nightly (26438b473 2021-01-08)
```
and 
```
rustc --version
rustc 1.51.0-nightly (c97f11af7 2021-01-10)
```

I don't doubt the existence of the bug, the  llvm asm backend has historically been flaky. Just wanted to check if there is an workaround we could do. We'll probably have to report it to rust/llvm too.

**Comment by gz** [comment]
Can reproduce with RUSTFLAGS="-Clink-dead-code"

**Comment by gz** [comment]
Thanks for reporting, I released a new version 0.35 which should fix your issue.

These function should not have been included in for the 64bit artifact anyways so I explicitly disabled them for this target.
